--- origsrc/pocl-0.13/lib/CL/Makefile.am	2016-04-04 08:45:46.000000000 -0500
+++ src/pocl-0.13/lib/CL/Makefile.am	2016-04-12 12:57:24.274613900 -0500
@@ -148,7 +148,7 @@ libpocl_la_SOURCES = clCreateContextFrom
 
 
 libpocl_la_CPPFLAGS = -I$(top_srcdir)/fix-include -I$(top_srcdir)/fix-include/OpenCL -I$(top_srcdir)/include -I$(top_srcdir)/lib/CL/devices $(OCL_ICD_CFLAGS)
-libpocl_la_LDFLAGS = -lltdl @PTHREAD_CFLAGS@ -version-info ${LIB_VERSION}
+libpocl_la_LDFLAGS = -lltdl @PTHREAD_CFLAGS@ -no-undefined -version-info ${LIB_VERSION}
 #libpocl_la_LDFLAGS += -Wl,-z,defs 
 
 if TCE_AVAILABLE
--- origsrc/pocl-0.13/lib/llvmopencl/Makefile.am	2016-04-04 08:45:46.000000000 -0500
+++ src/pocl-0.13/lib/llvmopencl/Makefile.am	2016-04-12 13:32:53.191952000 -0500
@@ -35,7 +35,7 @@ pkglib_LTLIBRARIES = llvmopencl.la
 
 AM_CXXFLAGS = -I@top_srcdir@/fix-include -I@top_srcdir@/include @LLVM_CXX_FLAGS@
 
-AM_LDFLAGS = -module -export-dynamic @LLVM_LDFLAGS@ -version-info ${KERNEL_COMPILER_LIB_VERSION}
+AM_LDFLAGS = -module -no-undefined -export-dynamic @LLVM_LIBS@ -avoid-version -shrext .so
 
 PASSES_SOURCES = Barrier.h						\
                         BarrierBlock.h BarrierBlock.cc				\
--- origsrc/pocl-0.13/lib/poclu/Makefile.am	2016-04-04 08:45:46.000000000 -0500
+++ src/pocl-0.13/lib/poclu/Makefile.am	2016-04-12 13:28:58.068595100 -0500
@@ -28,7 +28,7 @@ lib_LTLIBRARIES = libpoclu.la
 libpoclu_la_SOURCES = bswap.c misc.c cl_half.c
 
 libpoclu_la_CPPFLAGS = -I$(top_srcdir)/fix-include -I$(top_srcdir)/include 
-libpoclu_la_LDFLAGS = -lm -version-info ${LIB_VERSION}
+libpoclu_la_LDFLAGS = -lm -no-undefined -version-info ${LIB_VERSION}
 libpoclu_la_LIBADD = @OPENCL_LIBS@
 
 EXTRA_DIST = CMakeLists.txt
